common/include/pcl/console/print.h

```diff
@@ -118,6 +132,39 @@ namespace pcl
       L_VERBOSE
     };
 
+    struct LogRecord {
+      VERBOSITY_LEVEL level;
+      std::string message;
+    };
+
+    class LogRecorder {
+    public:
+      template <class T>
+      LogRecorder&
+      operator<<(const T& x)
+      {
+        s << x;
+
+        return *this;
+      }
+
+      LogRecorder&
+      operator<<(std::ostream& (std::ostream&))
```

Comment:
This one looks strange, what is it for?

Reply:
It was for handling std::endl if I remember correctly - that apparently need this operator.